lib/substitutions.js

```diff
@@ -178,10 +178,15 @@ async function getDisplayMap(main, room_id) {
     let users = await main.listGhostUsers(room_id);
     users = await Promise.all(users.map(user => store.select({id: user})));
     const displaymap = {};
+    users = users.filter((u) => u && u[0]);
     users.forEach(user => {
-        if (user && user[0]) {
-            displaymap[user[0].display_name] = user[0].id.split("_")[2].split(":")[0];
+        // The format is @slack_domain_userid:homeserver_domain
+        const localpart = user[0].id.split(":")[0].replace("@slack", "").replace("@_slack", "");
```

Comment:
is the prefix here not customisable? I thought this was a variable on main somewhere?

Reply:
...Yes you are right. For whatever reason this function isn't aware of that

Reply:
It could be though? I thought these had a reference to main?

Reply:
You should use [`main.getUserIDPrefix`](https://github.com/matrix-org/matrix-appservice-slack/blob/master/lib/Main.js#L206) here right?